sourced/ml/tests/test_token_parser.py

```diff
@@ -53,6 +53,49 @@ def test_process_token(self):
             res = list(self.tp.process_token(token))
             self.assertEqual(res, correct)
 
+    def test_process_token_with_attach_upper(self):
+        tp = TokenParser(stem_threshold=100, single_shot=True, max_token_length=100,
+                         min_split_length=1)
+        tokens = [
+            ("UpperCamelCase", ["upper", "camel", "case"]),
+            ("camelCase", ["camel", "case"]),
+            ("FRAPScase", ["frap", "scase"]),
+            ("SQLThing", ["sql", "thing"]),
+            ("_Astra", ["astra"]),
+            ("CAPS_CONST", ["caps", "const"]),
+            ("_something_SILLY_", ["something", "silly"]),
+            ("blink182", ["blink"]),
+            ("FooBar100500Bingo", ["foo", "bar", "bingo"]),
+            ("Man45var", ["man", "var"]),
+            ("method_name", ["method", "name"]),
+            ("Method_Name", ["method", "name"]),
+            ("101dalms", ["dalms"]),
+            ("101_dalms", ["dalms"]),
+            ("101_DalmsBug", ["dalms", "bug"]),
+            ("101_Dalms45Bug7", ["dalms", "bug"]),
+            ("wdSize", ["wd", "size"]),
+            ("Glint", ["glint"]),
+            ("foo_BAR", ["foo", "bar"]),
+            ("sourced.ml.algorithms.uast_ids_to_bag",
+             ["sourced", "ml", "algorithms", "uast", "ids", "to", "bag"]),
+            ("WORSTnameYOUcanIMAGINE", ["wors", "tname", "yo", "ucan", "imagine"]),
+            # Another bad example. Parser failed to parse it correctly
+            ("SmallIdsToFoOo", ["small", "ids", "to", "fo", "oo"]),
+            ("SmallIdFooo", ["small", "id", "fooo"]),
+            ("ONE_M0re_.__badId.example", ["one", "m", "re", "bad",
+                                           "id", "example"]),
+            ("never_use_Such__varsableNames", ["never", "use", "such", "varsable", "names"]),
+            ("a.b.c.d", ["a", "b", "c", "d"]),
+            ("A.b.Cd.E", ["a", "b", "cd", "e"]),
+            ("looong_sh_loooong_sh", ["looong", "sh", "loooong", "sh"]),
+            ("sh_sh_sh_sh", ["sh", "sh", "sh", "sh"]),
+            ("loooong_loooong_loooong", ["loooong", "loooong", "loooong"])
```

Comment:
Let's add `ONLYCAPS` and `nocaps` tokens to both tests to see that they remain the same.